### PR TITLE
protection for null pointer in snull

### PR DESCRIPTION
**Pull request by jypan-ccu**
Set protection for null pointer in snull, and modify snull_tx_timeout  for reset packet pool.

### CODE REVIEW

**Comment by jypan-ccu** [comment]
Apply proc_ops_wrapper( ) to scull* and change few down_interruptible( ) to mutex_lock_interruptible( ) in scull*

File: scull/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Comment/uncomment the following line to disable/enable debugging
-#DEBUG = y
```
Comment (dwalkes) [inline scull/Makefile:2]: @jypan-ccu thanks for the PR!

Did you intend to turn all of the DEBUG flags on?  My thought is that we leave them off by default and users can turn them on if they need them for debugging.

Comment (jypan-ccu) [reply scull/Makefile]: That's OK! I just make sure all the code segment works under DEBUG flags on. We should leave them off by default.
I have updated the pull request.

**Comment by dwalkes** [comment]
Thanks for your contribution!

**Pull request merged**